pkg/mapper/mapper.go

```diff
@@ -206,19 +211,42 @@ func (m *MetricMapper) InitFromYAMLString(fileContents string) error {
 	if m.MappingsCount != nil {
 		m.MappingsCount.Set(float64(len(n.Mappings)))
 	}
-
 	return nil
 }
 
-func (m *MetricMapper) InitFromFile(fileName string) error {
+func (m *MetricMapper) InitFromFile(fileName string, cacheSize int) error {
 	mappingStr, err := ioutil.ReadFile(fileName)
 	if err != nil {
 		return err
 	}
-	return m.InitFromYAMLString(string(mappingStr))
+
+	return m.InitFromYAMLString(string(mappingStr), cacheSize)
+}
+
+func (m *MetricMapper) InitCache(cacheSize int) {
+	m.useCache = true
+	if cacheSize == 0 {
+		m.useCache = false
+	} else {
+		cache, err := NewMetricMapperCache(cacheSize)
+		if err != nil {
+			log.Warnf("Unable to setup metric cache. Performance may be negatively impacted. Caused by: %s", err)
```

Comment:
Following the bread crumbs, the only error that could happen here would be [invalid configuration](https://github.com/hashicorp/golang-lru/blob/master/simplelru/lru.go#L28). I would prefer to

* prevent negative cache sizes ourselves
* if we fail to initialize the cache for another reason, fail hard

I'm not a fan of silent failures, even if it's something "optional" like a cache.

Reply:
Aside from that, this is purely a matter of taste (so no need to change it if you disagree): I'm not a fan of "if use $feature …" – could we install a NOOP cache instead?

Reply:
Happy to do so! I should have followed the breadcrumbs more carefully myself.